frontend/src/components/pages/topics/QuickInfo.tsx

```diff
@@ -42,19 +40,30 @@ export const TopicQuickInfoStatistic = observer((p: { topic: Topic }) => {
     statsAr.push(<Statistic key="msgs" title="Messages" value={messageSum} />);
 
     // Config Entries / Seperator
-    const configEntries = filterTopicConfig(api.topicConfig.get(topic.topicName)?.configEntries);
-    if (configEntries) {
-        const configStats = uiState.topicSettings.favConfigEntries
-            .map(favName => configEntries!.find(e => e.name === favName))
+    const configEntries = api.topicConfig.get(topic.topicName)?.configEntries;
+    const filteredConfigEntries = filterTopicConfig(configEntries);
+    const cleanupPolicy = configEntries?.find(x => x.name == 'cleanup.policy')?.value;
+    if (configEntries && filteredConfigEntries && cleanupPolicy) {
+        const dynamicEntries = ['cleanup.policy'];
+
+        if (cleanupPolicy.includes('delete')) {
+            dynamicEntries.push('retention.ms');
+            dynamicEntries.push('retention.bytes');
+        }
+        if (cleanupPolicy.includes('compact')) {
+            dynamicEntries.push('segment.ms');
+            dynamicEntries.push('segment.bytes');
+        }
+
```

Comment:
[sand]: I see this and I notice we could reuse this logic in the future, not blocking at all, it's very easy to read and understand, but In my mind a helper function would be ideal, so you could potentially reuse it.

Reply:
Where would we reuse it?
What would the helper function do? 

Are you thinking about something like extracting name+value pairs of the config?
Or the logic where it switches what kinds of configs to display depending on cleanupPolicy?